pkg/daemon/daemon.go

```diff
@@ -303,86 +308,108 @@ func ptp4lCreateCmd(nodeProfile *ptpv1.PtpProfile, confFilePath string) *exec.Cm
 	return exec.Command(args[0], args[1:]...)
 }
 
-// cmdRun runs given ptpProcess and wait for errors
+// cmdRun runs given ptpProcess and restarts on errors
 func cmdRun(p *ptpProcess, stdoutToSocket bool) {
-	glog.Infof("Starting %s...", p.name)
-	glog.Infof("%s cmd: %+v", p.name, p.cmd)
 	var c net.Conn
-	var err error
-	defer func() {
-		if stdoutToSocket && c != nil {
-			if err := c.Close(); err != nil {
-				glog.Errorf("closing connection returned error %s", err)
-			}
+	done := make(chan struct{}) // Done setting up logging.  Go ahead and wait for process
+	for {
+		glog.Infof("Starting %s...", p.name)
+		glog.Infof("%s cmd: %+v", p.name, p.cmd)
+
+		//
+		// don't discard process stderr output
+		//
+		p.cmd.Stderr = os.Stderr
+		cmdReader, err := p.cmd.StdoutPipe()
+		if err != nil {
+			glog.Errorf("cmdRun() error creating StdoutPipe for %s: %v", p.name, err)
+			break
 		}
-		p.exitCh <- true
-	}()
-
-	//
-	// don't discard process stderr output
-	//
-	p.cmd.Stderr = os.Stderr
-	cmdReader, err := p.cmd.StdoutPipe()
-	if err != nil {
-		glog.Errorf("cmdRun() error creating StdoutPipe for %s: %v", p.name, err)
-		return
-	}
-
-	done := make(chan struct{})
-
-	if !stdoutToSocket {
-		scanner := bufio.NewScanner(cmdReader)
-		go func() {
-			for scanner.Scan() {
-				output := scanner.Text()
-				fmt.Printf("%s\n", output)
-				extractMetrics(p.configName, p.name, p.ifaces, output)
-			}
-			done <- struct{}{}
-		}()
-	} else {
-		go func() {
-		connect:
-			select {
-			case <-p.exitCh:
-				done <- struct{}{}
-			default:
-				c, err = net.Dial("unix", eventSocket)
-				if err != nil {
-					glog.Errorf("error trying to connect to event socket")
-					time.Sleep(connectionRetryInterval)
-					goto connect
-				}
-			}
+		if !stdoutToSocket {
 			scanner := bufio.NewScanner(cmdReader)
-			for scanner.Scan() {
-				output := scanner.Text()
-				out := fmt.Sprintf("%s\n", output)
-				fmt.Printf("%s", out)
-				_, err := c.Write([]byte(out))
-				if err != nil {
-					glog.Errorf("Write error:", err)
-					goto connect
+			go func() {
+				for scanner.Scan() {
+					output := scanner.Text()
+					fmt.Printf("%s\n", output)
+					extractMetrics(p.configName, p.name, p.ifaces, output)
+				}
+				done <- struct{}{}
+			}()
+		} else {
+			go func() {
```

Comment:
every time you go over a restart, won't this spawn a  new go routine and the old one will still be running and trying to reconnect?